spec/controllers/sessions_controller_spec.rb

```diff
@@ -0,0 +1,82 @@
+require 'rails_helper'
+
+describe SessionsController do
+
+  describe 'Get new' do
+    context 'with authenticated user' do 
+      it 'renders the new template' do 
+        session[:user_id] = nil
+        get :new
+        expect(response).to render_template :new
+      end
+    end
+    context 'with unauthenticated user' do
+      it 'should redirect to home_path' do 
+        session[:user_id] = Fabricate(:user).id
+        get :new
+        expect(response).to redirect_to home_path
+      end
+    end
+  end
+
+  describe 'Post create' do
+    context 'with valid credentials' do
+      before do
+        @thupten = Fabricate(:user)
+        post :create, email: @thupten.email, password: @thupten.password
+      end 
+
+      it 'sets sessions[:user_id]' do
+        expect(session[:user_id]).to eq(@thupten.id)
+      end
+
+      it 'redirects to home_path after successful login' do
+        expect(response).to redirect_to home_path
+      end
+
+      it 'sets the notice' do
+        expect(flash[:notice]).not_to be_blank
+      end
+    end
+
+    context 'with invalid credentials' do
+      before do
+        thupten = Fabricate(:user)
```

Comment:
I would recommend using `let(:thupten) { Fabricate(:user) }`, `let` allows you to lazy load and define variables for your specs. You have to use it outside the `before` block though.